processor/spanmetricsprocessor/processor.go

```diff
@@ -382,61 +375,44 @@ func (p *processorImp) aggregateMetricsForSpan(serviceName string, span ptrace.S
 		latencyInMilliseconds = float64(endTime-startTime) / float64(time.Millisecond.Nanoseconds())
 	}
 
-	// Binary search to find the latencyInMilliseconds bucket index.
-	index := sort.SearchFloat64s(p.latencyBounds, latencyInMilliseconds)
-
 	key := buildKey(serviceName, span, p.dimensions, resourceAttr)
 
 	p.cache(serviceName, span, key, resourceAttr)
-	p.updateCallMetrics(key)
-	p.updateLatencyMetrics(key, latencyInMilliseconds, index)
-	p.updateLatencyExemplars(key, latencyInMilliseconds, span.TraceID(), span.SpanID())
-}
-
-// updateCallMetrics increments the call count for the given metric key.
-func (p *processorImp) updateCallMetrics(key metricKey) {
-	p.callSum[key]++
+	p.updateHistogram(key, latencyInMilliseconds, span.TraceID(), span.SpanID())
 }
 
 // resetAccumulatedMetrics resets the internal maps used to store created metric data. Also purge the cache for
 // metricKeyToDimensions.
 func (p *processorImp) resetAccumulatedMetrics() {
-	p.callSum = make(map[metricKey]int64)
-	p.latencyCount = make(map[metricKey]uint64)
-	p.latencySum = make(map[metricKey]float64)
-	p.latencyBucketCounts = make(map[metricKey][]uint64)
+	p.histograms = make(map[metricKey]*histogramData)
 	p.metricKeyToDimensions.Purge()
 }
 
-// updateLatencyExemplars sets the histogram exemplars for the given metric key and append the exemplar data.
-func (p *processorImp) updateLatencyExemplars(key metricKey, value float64, traceID pcommon.TraceID, spanID pcommon.SpanID) {
-	if _, ok := p.latencyExemplarsData[key]; !ok {
-		p.latencyExemplarsData[key] = []exemplarData{}
+// updateHistogram sets the histogram exemplars for the given metric key and append the exemplar data.
```

Comment:
I think this comment needs to be updated since it's no longer specific to updating exemplars.

Reply:
Done.